### ISSUE TITLE
vignette computation fails

### ISSUE THREAD

**Issue by vjcitn** [snaketron/IgGeneUsage#10]
Hi -- the chunks in the vignette are all `eval=F`?  This prevents the Bioconductor build system from
realistically checking the package.  I tried running the vignette (changing eval=F to eval=TRUE,
and using IgGeneUsage:::DGU because DGU function is not exported) and after two chains finish
we have:
```
Computing summaries ... 

Posterior extraction ... 

Computing probability of DGU ... 

Computing posterior predictions ... 

Error: $ operator is invalid for atomic vectors
In addition: Warning messages:
1: Bulk Effective Samples Size (ESS) is too low, indicating posterior means and medians may be unreliable.
Running the chains for more iterations may help. See
https://mc-stan.org/misc/warnings.html#bulk-ess 
2: Tail Effective Samples Size (ESS) is too low, indicating posterior variances and tail quantiles may be unreliable.
Running the chains for more iterations may help. See
https://mc-stan.org/misc/warnings.html#tail-ess 

Enter a frame number, or 0 to exit   

1: source("IgUsageCaseStudies.R", echo = TRUE)
2: withVisible(eval(ei, envir))
3: eval(ei, envir)
4: eval(ei, envir)
5: IgUsageCaseStudies.R#64: IgGeneUsage:::DGU(usage.data = IGHV_HCV, mcmc.warm
6: getPpcRepertoire(glm = glm, usage.data = usage.data, hdi.level = hdi.level)
```
```
> sessionInfo()
R version 4.2.0 RC (2022-04-15 r82209)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /home/stvjc/R-4-2-dist/lib/R/lib/libRblas.so
LAPACK: /home/stvjc/R-4-2-dist/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] reshape2_1.4.4              rstan_2.21.5               
 [3] ggrepel_0.9.1               gridExtra_2.3              
 [5] ggforce_0.3.3               ggplot2_3.3.6              
 [7] IgGeneUsage_1.11.1          StanHeaders_2.21.0-7       
 [9] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[11] GenomicRanges_1.48.0        GenomeInfoDb_1.32.2        
[13] IRanges_2.30.0              S4Vectors_0.34.0           
[15] BiocGenerics_0.42.0         MatrixGenerics_1.8.0       
[17] matrixStats_0.62.0          Rcpp_1.0.8.3               
[19] knitr_1.39                  rmarkdown_2.14             

loaded via a namespace (and not attached):
 [1] lattice_0.20-45        prettyunits_1.1.1      ps_1.7.0              
 [4] assertthat_0.2.1       digest_0.6.29          utf8_1.2.2            
 [7] plyr_1.8.7             R6_2.5.1               evaluate_0.15         
[10] highr_0.9              pillar_1.7.0           zlibbioc_1.42.0       
[13] rlang_1.0.2            callr_3.7.0            Matrix_1.4-1          
[16] startup_0.18.0         labeling_0.4.2         stringr_1.4.0         
[19] loo_2.5.1              RCurl_1.98-1.6         polyclip_1.10-0       
[22] munsell_0.5.0          DelayedArray_0.22.0    compiler_4.2.0        
[25] xfun_0.31              pkgconfig_2.0.3        pkgbuild_1.3.1        
[28] htmltools_0.5.2        tidyselect_1.1.2       tibble_3.1.7          
[31] GenomeInfoDbData_1.2.8 codetools_0.2-18       fansi_1.0.3           
[34] crayon_1.5.1           dplyr_1.0.9            withr_2.5.0           
[37] MASS_7.3-57            bitops_1.0-7           grid_4.2.0            
[40] gtable_0.3.0           lifecycle_1.0.1        DBI_1.1.2             
[43] magrittr_2.0.3         scales_1.2.0           RcppParallel_5.1.5    
[46] cli_3.3.0              stringi_1.7.6          farver_2.1.0          
[49] XVector_0.36.0         ellipsis_0.3.2         generics_0.1.2        
[52] vctrs_0.4.1            tools_4.2.0            glue_1.6.2            
[55] tweenr_1.0.2           purrr_0.3.4            parallel_4.2.0        
[58] processx_3.6.0         fastmap_1.1.0          inline_0.3.19         
[61] colorspace_2.0-3  
```

**Comment by snaketron** [comment]
Hi,
the fact MCMC sampling was done tells me that function DGU is exported. The problem must be something else. Can you upload a subset of your input or at least describe it so that I can figure out what the problem might be.

`
Computing summaries ... 
Posterior extraction ... 
Computing probability of DGU ... 
Computing posterior predictions ... 
`


> the chunks in the vignette are all eval=F? This prevents the Bioconductor build system from realistically checking the package.
You are right about this. When the package was first accepted to Bioconductor the evals were set to TRUE. We disabled them only for the last release and will set  them back to TRUE for the next one.  BTW: the code was not modified severely between the releases.

**Comment by vjcitn** [comment]
I don't have data, I am just trying to run the vignette after setting eval=TRUE on the chunks (and I used IgGeneUsage:::DGU to deal with the export oddity).  Let me try again with a proper checkout.

**Comment by vjcitn** [comment]
Runs well with a correct checkout.  Looking forward to a fully evaluated vignette.
